packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -170,6 +171,27 @@ export interface IProject extends IResource, iam.IGrantable, ec2.IConnectable {
    * @default sum over 5 minutes
    */
   metricFailedBuilds(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Defines a Codestar notification rule triggered when the build project events occurs by you specified.
+   * It very similar to `onEvent` API
+   * You can also use the methods `notifyOnBuildSucceeded` and `notifyOnBuildFailed` to define rules for these specific event occurs.
+   *
+   * @param id The id of the Codestar notification rule
+   * @param options Customization options for Codestar notification rule
+   * @returns Codestar notification rule associated with this build project.
+   */
+  notifyOnEvent(id: string, options?: notifications.NotifyOnEventOptions): notifications.Rule;
+
+  /**
+   * Defines a Codestar notification rule which triggers when a build completes successfully.
+   */
+  notifyOnBuildSucceeded(id: string, options?: notifications.NotifyOnEventOptions): notifications.Rule;
```

Comment:
Having `options?: notifications.NotifyOnEventOptions` here doesn't really make sense, because `NotifyOnEventOptions` contain the `events` property, but the event types for `notifyOnBuildSucceeded` are fixed - you cannot pass different ones for them.

We need a new interface in the `@aws-cdk/aws-codestarnotifications` package that is today's `NotifyOnEventOptions`, except without `events`. Then have `NotifyOnEventOptions` extend that new interface, and add `events`.

Same comment in the other methods in this PR.